src/Windows/Avalonia.Direct2D1/SwapChainRenderTarget.cs

```diff
@@ -69,9 +58,25 @@ public IDrawingContextImpl CreateDrawingContext(IVisualBrushRenderer visualBrush
                 visualBrushRenderer,
                 _deviceContext,
                 DirectWriteFactory,
+                WicImagingFactory,
                 _swapChain);
         }
 
+        public IRenderTargetBitmapImpl CreateLayer(int pixelWidth, int pixelHeight)
```

Comment:
What happens on window resize? According to https://github.com/AvaloniaUI/Avalonia/blob/de7ef1bca4e52c6ff0ec956c973b9b75e86b0d0a/src/Windows/Avalonia.Direct2D1/SwapChainRenderTarget.cs#L52 _deviceContext will be recreated. But we still have layers associated with it.

There is also a concern about switching graphics device (e. g. when user connects through RDP) - layers won't be compatible anymore.

Reply:
This still seems to work even when `_deviceContext` is recreated; resize works fine.

I did actually try switching graphics device (by disabling my gfx card while running) but it seems we don't handle it properly elsewhere, and everything freezes anyway, so I was going to address that separately.